pkg/app/middleware.go

```diff
@@ -21,10 +23,19 @@ func WithMetricsResponderMiddleware(next http.Handler) http.Handler {
 }
 
 // WithHealthMiddleware returns middleware which responds to the /health endpoint
-func WithHealthMiddleware(next http.Handler) http.Handler {
+func WithHealthMiddleware(next http.Handler, threads []*thread.Thread, logger logrus.FieldLogger, timeout time.Duration) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.Method == http.MethodGet && r.URL.Path == "/health" {
-			w.WriteHeader(http.StatusOK)
+			status := http.StatusOK
+			for _, th := range threads {
+				if th.LastRunTimestamp().IsZero() && time.Since(th.LastRunTimestamp()) > timeout {
```

Comment:
Should it be or?

Reply:
Actually it doesn't matter. if it's not zero then we have something and the second part will do the work

Reply:
OK no it should be OR

Reply:
https://play.golang.org/p/U37w8T5p2lb

Reply:
Should be 
`if !th.LastRunTimestamp().IsZero() && time.Since(th.LastRunTimestamp()) > timeout`
Fixed

Reply:
A comment related to https://github.com/openshift/assisted-service/pull/2619#discussion_r708166865 
I know that right now it will work but if we already there so maybe we should avoid `IsZero`?